src/otx/algo/detection/yolov9.py

```diff
@@ -0,0 +1,192 @@
+    mean: tuple[float, float, float] = (0.0, 0.0, 0.0)
+    std: tuple[float, float, float] = (255.0, 255.0, 255.0)
+
+    def __init__(
+        self,
+        model_name: Literal["yolov9_s", "yolov9_m", "yolov9_c"],
+        label_info: LabelInfoTypes,
+        input_size: tuple[int, int] = (640, 640),
+        optimizer: OptimizerCallable = DefaultOptimizerCallable,
+        scheduler: LRSchedulerCallable | LRSchedulerListCallable = DefaultSchedulerCallable,
+        metric: MetricCallable = MeanAveragePrecisionFMeasureCallable,
+        torch_compile: bool = False,
+        tile_config: TileConfig = TileConfig(enable_tiler=False),
+    ) -> None:
+        self.load_from: str = PRETRAINED_WEIGHTS[model_name]
+        super().__init__(
+            model_name=model_name,
+            label_info=label_info,
+            input_size=input_size,
+            optimizer=optimizer,
+            scheduler=scheduler,
+            metric=metric,
+            torch_compile=torch_compile,
+            tile_config=tile_config,
+        )
+
+    def _build_model(self, num_classes: int) -> SingleStageDetector:
+        backbone = GELAN(model_name=self.model_name)
+        neck = YOLONeck(model_name=self.model_name)
+        bbox_head = YOLOHead(model_name=self.model_name, num_classes=num_classes)
+
+        detector = SingleStageDetector(
+            backbone=backbone,
+            neck=neck,
+            bbox_head=bbox_head,
+            criterion=None,
+        )
+        # patch _load_from_state_dict
+        detector._load_from_state_dict = _load_from_state_dict_for_yolov9.__get__(detector, SingleStageDetector)  # type: ignore[method-assign] # noqa: SLF001
+
+        # set criterion
+        strides: list[int] | None = [8, 16, 32] if self.model_name == "yolov9_c" else None
+        self.vec2box = Vec2Box(detector, self.input_size, strides)
+        detector.bbox_head.vec2box = self.vec2box
+        detector.criterion = YOLOv9Criterion(
+            num_classes=num_classes,
+            loss_cls=BCELoss(),
+            loss_dfl=DFLoss(self.vec2box),
+            loss_iou=BoxLoss(),
+            vec2box=self.vec2box,
+        )
+
+        return detector
+
+    @property
+    def _exporter(self) -> OTXModelExporter:
+        """Creates OTXModelExporter object that can export the model."""
+        if self.input_size is None:
+            msg = f"Input size attribute is not set for {self.__class__}"
+            raise ValueError(msg)
+
+        if self.explain_mode:
+            msg = "Explainable model export is not supported for YOLOv9 yet."
+            raise NotImplementedError(msg)
+
+        return OTXNativeModelExporter(
+            task_level_export_parameters=self._export_parameters,
+            input_size=(1, 3, *self.input_size),
+            mean=self.mean,
+            std=self.std,
+            resize_mode="fit_to_window_letterbox",
+            pad_value=114,
+            swap_rgb=True,
+            via_onnx=True,
```

Comment:
Are there any errors using direct OpenVINO conversion tool?

Reply:
I tried with openvino 2024.3, but got the below error:
```shell
  File "venv/otx_v2/lib/python3.11/site-packages/model_api/models/ssd.py", line 121, in __init__
    self.labels_layer = find_layer_by_name(labels_layer, layers)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "venv/otx_v2/lib/python3.11/site-packages/model_api/models/ssd.py", line 82, in find_layer_by_name
    suitable_layers = [layer_name for layer_name in layers if name in layer_name]
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "venv/otx_v2/lib/python3.11/site-packages/model_api/models/ssd.py", line 82, in <listcomp>
    suitable_layers = [layer_name for layer_name in layers if name in layer_name]
                                                              ^^^^^^^^^^^^^^^^^^
TypeError: argument of type 'openvino._pyopenvino.Output' is not iterable
```

It looks that output name is not set and actually output name is not inserted into `openvino.convert_model`.
https://github.com/openvinotoolkit/training_extensions/blob/709d841f4a2c9e8dc73b5e416a1deb7536a6b761/src/otx/core/exporter/native.py#L85-L89

I re-tried it with `output=["boxes", "labels"] as an argument like below,
```python
exported_model = openvino.convert_model(
    model,
    example_input=dummy_tensor,
    input=(openvino.runtime.PartialShape(self.input_size),),
    output=["boxes", "labels"],
)
```

but another error occurred:
```shell
  File "venv/otx_v2/lib/python3.11/site-packages/openvino/tools/ovc/convert_impl.py", line 494, in _convert
    ov_model = driver(argv, {"conversion_parameters": non_default_params})
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "venv/otx_v2/lib/python3.11/site-packages/openvino/tools/ovc/convert_impl.py", line 250, in driver
    ov_model = moc_emit_ir(prepare_ir(argv), argv)
                           ^^^^^^^^^^^^^^^^
  File "venv/otx_v2/lib/python3.11/site-packages/openvino/tools/ovc/convert_impl.py", line 194, in prepare_ir
    ov_model = moc_pipeline(argv, moc_front_end)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "venv/otx_v2/lib/python3.11/site-packages/openvino/tools/ovc/moc_frontend/pipeline.py", line 136, in moc_pipeline
    oplaces.append(out_desc["name"])
                   ~~~~~~~~^^^^^^^^
KeyError: 'name'
```

It seems that `fe_output_user_data_repack` return the list of dicts that don't have `name` key.
It seems openvino issue that should be escalated and after resolving this, we can try openvino conversion without onnx pipeline again.
https://github.com/openvinotoolkit/openvino/blob/22079ca11d69de2721c11502db9d922b362e15be/tools/ovc/openvino/tools/ovc/moc_frontend/extractor.py#L308-L339

Reply:
Made an issue https://github.com/openvinotoolkit/openvino/issues/26457